### ISSUE TITLE
Bug: Tomb `Wait` blocks forever when no Goroutine is ever spawned.

### ISSUE THREAD

**Issue by jpittis** [go-tomb/tomb#21]
If you call `tmb.Wait()` on a tomb that has never called `tmb.Go`, it blocks forever.

Here's a failing test case to reproduce the bug:

````go
package tomb_test

import (
	"testing"
	"time"

	tomb "gopkg.in/tomb.v2"
)

func TestTombWaitDoesNotBlockWhenNoGoRoutinesSpawned(t *testing.T) {
	var tmb tomb.Tomb

	tmb.Kill(nil)

	done := make(chan struct{})
	go func() {
		err := tmb.Wait()
		if err != nil {
			t.Fatal(err)
		}
		close(done)
	}()

	select {
	case <-time.After(5 * time.Second):
		t.Fatal("Tomb did not exit in time.")
	case <-done:
		// Success!
	}
}
````

Here's the output of the test case using tomb d5d1b5820637886def9eef33e03a27a9f166942c:

````
$ go test
--- FAIL: TestTombWaitDoesNotBlockWhenNoGoRoutinesSpawned (5.00s)
	tomb_test.go:26: Tomb did not exit in time.
FAIL
exit status 1
FAIL	github.com/jpittis/tombbug	5.008s
````

**Comment by niemeyer** [comment]
That's by design. It prevents the potential race between Wait and Go. The proper way to use a tomb is to never hand it off for someone to Wait on it unless you do intend to use it.

**Comment by jpittis** [comment]
Appreciate the fast reply. Obviously there was another issue opened with the same subject.

Sorry about create a duplicate.
